### ISSUE TITLE
403 Forbidden

### ISSUE THREAD

**Issue by richlock1** [lostfields/python-panasonic-comfort-cloud#33]
Hi,

From today I have this problem when I want to use this module or the sample application:

./pcomfortcloud.py -r True <username> <password> list
--- creating token by authenticating
Traceback (most recent call last):
  File "./pcomfortcloud.py", line 5, in <module>
    __main__.main()
  File "/home/ricsi/Downloads/trees/pcomfort/python-panasonic-comfort-cloud/pcomfortcloud/__main__.py", line 202, in main
    session.login()
  File "/home/ricsi/Downloads/trees/pcomfort/python-panasonic-comfort-cloud/pcomfortcloud/session.py", line 99, in login
    self._create_token()
  File "/home/ricsi/Downloads/trees/pcomfort/python-panasonic-comfort-cloud/pcomfortcloud/session.py", line 131, in _create_token
    raise ResponseError(response.status_code, response.text)
pcomfortcloud.session.ResponseError: Invalid response, status code: 403 - Data: <html>
<head><title>403 Forbidden</title></head>
<body>
<center><h1>403 Forbidden</h1></center>
<hr><center>nginx</center>
</body>
</html>

It worked fine before for several months. I have tried it from other LAN also, the same problem. Tried with other username also, same problem. Tried -t and -s switches, no success. 

I am using the latest version of this module:

commit d9eeaf0ec4d16f5ea3bc90275588c3dd802b4f67 (HEAD -> master, origin/master, origin/HEAD)
Author: lostfields <nicolai.gjaerum@gmail.com>
Date:   Sun Sep 20 09:58:19 2020 +0200

    fixing #32 by urlencode

Do you have the same experience, or is this only my problem?

The android app is still working well, but with that I cannot automate the ACs.

Thanks in advance.

**Comment by TomS1985** [comment]
same problem here.

**Comment by lostfields** [comment]
Hi @richlock1 and @TomS1985 ! Before we take this any further, may you check if you have any new agreement license you have to accept for the current login you are using?

**Comment by richlock1** [comment]
Yes, I have accepted the new license agreement with android app, but still I have this problem.
Do you have also similar problem?
Could it be related specific devices, like this python script? Maybe some signature should be updated in this script...
You know it better.

**Comment by TomS1985** [comment]
I have also accepted the agreement license with my android app (because I know this "problem" from the past). But this time, I cannot read the values with the perl script anyway.

**Comment by lostfields** [comment]
It should be fixed now, I'll try to upload new version to PyPi once I figure out my password etc for my new laptop

**Comment by richlock1** [comment]
Thank you for the really fast correction. It is working fine again :)
Did you sniff the android app to identify what to add in HTML header?

Do you think if it is feasible to improve this fantastic library to work only with LAN so we can skip the cloud?

**Comment by lostfields** [comment]
I did sniff the android app, but the calls was exactly the same. My local calls in https://github.com/lostfields/python-panasonic-comfort-cloud/blob/37595976651450a35ff5a7e7989d658b2833c8e9/requests.http#L5 worked too.

Strange thing, but then I remembered that the VS Code request extension is adding User-Agent header. It worked when I added any User-Agent. Don't know if Panasonic is doing this kind of stuff to block others using their API or if there's some kind of statistics/optimizations they are doing

I would like a local LAN communication as well, but I think the unit itself is contacting the cloud, and commands is pushed down a direct communication line - that may be a lot harder, and you probably have to spoof the url the unit is contacting or flash the unit istelf. Don't think it's worth the try.
